src/_data/LiveData.js

```diff
@@ -172,36 +172,38 @@ const initAuthorized = async (authData, store) => {
                 allowedMarkets = landingCompany.financial_company.legal_allowed_markets;
             }
             canUpgrade = !hasAccountOfType('real', accounts);
-        } else if (landingCompany.financial_company.shortcode === 'maltainvest') {
-            typeOfNextAccount = 'financial';
-            canUpgrade = !hasAccountOfType('financial', accounts);
-            currencyOptions = landingCompany.financial_company.legal_allowed_currencies;
-            allowedMarkets = landingCompany.financial_company.legal_allowed_markets;
-        } else if (landingCompany.financial_company.shortcode === 'costarica') {
-            allowedMarkets = landingCompany.financial_company.legal_allowed_markets;
-            const legalAllowedCurrencies = landingCompany.financial_company.legal_allowed_currencies;
-            const existingCurrencies = getExistingCurrencies(accounts);
-            if (existingCurrencies.length) {
-                const dividedExistingCurrencies = groupCurrencies(existingCurrencies, currencyConfig);
-                if (dividedExistingCurrencies.fiatCurrencies.length) {
-                    const dividedAllowedCurrencies = groupCurrencies(legalAllowedCurrencies, currencyConfig);
-                    const legalAllowedCryptoCurrencies = dividedAllowedCurrencies.cryptoCurrencies;
-                    const existingCryptoCurrencies = dividedExistingCurrencies.cryptoCurrencies;
-                    currencyOptions = legalAllowedCryptoCurrencies.filter(x => existingCryptoCurrencies.indexOf(x) === -1);
-                    if (currencyOptions.length) {
+        } else if (landingCompany.financial_company) {
```

Comment:
@nazaninreihani in MLT user cannot set currency because unable to find `shortcode` of `undefined`, so I added this small check here, but I'm not sure if it will cover all use cases? It works fine here.

Reply:
hm. Should have made a new PR, but it is the same issue name so...

Reply:
@brucebinary you mean when you tried to upgrade from MLT to MF via create new account the select box of currency was empty?

Reply:
My bad for the confusion. The error happens when user that can only have MLT accounts (Belgium).

Reply:
@brucebinary I see it now, I think this PR will fix it, thanks